backend/core/auth/src/main/java/io/writeopia/api/core/auth/routing/CookieAuthRouting.kt

```
@@ -161,21 +162,36 @@ fun Routing.cookieAuthRoute(writeopiaDb: WriteopiaDbBackend, debugMode: Boolean
-            val expiresAt = parts.getOrNull(1)?.toLongOrNull()
+        if (accessToken != null) {
+            // Validate the JWT and extract userId from verified claims
+            // This prevents forged cookies from being accepted
+            val userId = JwtConfig.extractUserId(accessToken)
+
+            if (userId != null) {
+                // Token is valid - extract expiry from the verified JWT
+                val expiresAt = try {
+                    val jwt = JwtConfig.accessVerifier.verify(accessToken)
+                    jwt.expiresAt?.time
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)